src/mumble/OpusCodec.cpp

```diff
@@ -0,0 +1,92 @@
+// Copyright 2005-2018 The Mumble Developers. All rights reserved.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file at the root of the
+// Mumble source tree or at <https://www.mumble.info/LICENSE>.
+
+#include "mumble_pch.hpp"
+
+#include "OpusCodec.h"
+
+#include "Audio.h"
+#include "Version.h"
+#include "MumbleApplication.h"
+
+#ifdef Q_CC_GNU
+#define RESOLVE(var) { var = reinterpret_cast<__typeof__(var)>(qlOpus.resolve(#var)); bValid = bValid && (var != NULL); }
+#else
+#define RESOLVE(var) { * reinterpret_cast<void **>(&var) = static_cast<void *>(qlOpus.resolve(#var)); bValid = bValid && (var != NULL); }
+#endif
+
+OpusCodec::OpusCodec() {
+	bValid = false;
+	qlOpus.setLoadHints(QLibrary::ResolveAllSymbolsHint);
+
+	QStringList alternatives;
+#if defined(Q_OS_MAC)
+	alternatives << QString::fromLatin1("libopus0.dylib");
+	alternatives << QString::fromLatin1("opus0.dylib");
+	alternatives << QString::fromLatin1("libopus.dylib");
+	alternatives << QString::fromLatin1("opus.dylib");
+#elif defined(Q_OS_UNIX)
+	alternatives << QString::fromLatin1("libopus0.so");
+	alternatives << QString::fromLatin1("libopus.so");
+	alternatives << QString::fromLatin1("opus.so");
+#else
+	alternatives << QString::fromLatin1("opus0.dll");
+#endif
+	foreach(const QString &lib, alternatives) {
+		qlOpus.setFileName(MumbleApplication::instance()->applicationVersionRootPath() + QLatin1String("/") + lib);
+		if (qlOpus.load()) {
+			bValid = true;
+			break;
+		}
+
+#ifdef Q_OS_MAC
+		qlOpus.setFileName(QApplication::instance()->applicationDirPath() + QLatin1String("/../Codecs/") + lib);
+		if (qlOpus.load()) {
+			bValid = true;
+			break;
+		}
+#endif
+
+#ifdef PLUGIN_PATH
+		qlOpus.setFileName(QLatin1String(MUMTEXT(PLUGIN_PATH) "/") + lib);
```

Comment:
What is this? What is `PLUGIN_PATH`?

Reply:
Copied from `CeltCodec`, it looks for the library in the plugins folder.

Reply:
Our PA plugins? How does that make sense for either CELT or Opus?

Reply:
On Debian the directory is `/usr/lib/mumble`: https://packages.debian.org/stable/amd64/mumble/filelist

The positional audio plugins and all the libraries are stored there.

Reply:
I’m a little confused where `PLUGIN_PATH` is defined though. But ok.

Reply:
It's defined by the package maintainer.